woocommerce/woocommerce-ios
# [Coupons in Orders]Remove extra logic for coupons to be consistent with other order lines

Closes: #10000


## Description

With this PR we remove the extra logic for coupons that was blocking any interaction with the order lines in the editable order screen after a coupon was removed. 

While blocking these lines was a good thing, the logic to ascertain it was not: we were relying on the order discount total to be the same as the coupons total. This is wrong: an order can have another type of discount besides coupons. Because of that, we can be blocking the UI unintentionally.

Given that we do not block the order lines UI after items, shipping, and fees are removed, I made the coupons follow the same logic to be consistent, instead of trying to come up with a new logic that might imply a medium-sized refactor.

## Testing instructions


1. Go to orders
2. Tap to create a new order
3. Tap to add a coupon
4. Add a valid coupon. The lines UI is blocked.
5. Open the coupon line again
6. Remove the coupon
7. Check that the coupon is removed from the order, but the lines UI is not blocked.

### Known issue
If you open any order line detail screen to add a new one right away (and very fast) you have deleted them, the text field resets when the order finished syncing the deletion. This is known and was there before.

## Screenshots


https://github.com/woocommerce/woocommerce-ios/assets/1864060/68464ac8-854b-43f7-b78b-63c638b35cdc


---
